chebi-update/src/test/java/org/reactome/release/test/chebiupdate/ChebiUpdaterTest.java

```diff
@@ -70,92 +98,215 @@
+			when(molecule2.getDBID()).thenReturn(456L);
+			when(molecule2.getSchemClass()).thenReturn(refMolSchemaClass);
+			when(adaptor.fetchInstance(456L)).thenReturn(molecule2);
+			
+			when(molecule3.getAttributeValue("identifier")).thenReturn("445566");
+			
+			when(molecule4.getAttributeValue("identifier")).thenReturn("778899");
+			when(molecule4.getAttributeValuesList("name")).thenReturn(new ArrayList<String>(Arrays.asList("NAME1")));
+			when(molecule4.getAttributeValue("formula")).thenReturn("H2");
+			when(molecule4.getSchemClass()).thenReturn(refMolSchemaClass);
+			when(molecule4.getDBID()).thenReturn(444L);
+			when(adaptor.fetchInstance(444L)).thenReturn(molecule4);
+			
+			when(refEnt1.getAttributeValuesList("name")).thenReturn(new ArrayList<String>(Arrays.asList("OTHERNAME", "NAME1", "BLAH")));
+			when(refEnt1.getSchemClass()).thenReturn(simpleEntitySchemaClass);
+			when(refEnt2.getAttributeValuesList("name")).thenReturn(new ArrayList<String>(Arrays.asList("TEST", "NAME2")));
+			when(refEnt2.getSchemClass()).thenReturn(simpleEntitySchemaClass);
+			when(refEnt3.getAttributeValuesList("name")).thenReturn(new ArrayList<String>(Arrays.asList("NAME-2", "ASDFQ@#$FASFASDF","NOPE")));
+			when(refEnt1.getSchemClass()).thenReturn(chemicalDrugSchemaClass);
+			
+			when(refEnt1.getAttributeValue(ReactomeJavaConstants.created)).thenReturn(createdInstance);
+			when(refEnt2.getAttributeValue(ReactomeJavaConstants.created)).thenReturn(createdInstance);
+			when(refEnt3.getAttributeValue(ReactomeJavaConstants.created)).thenReturn(createdInstance);
+			
+			Collection<GKInstance> referrers1 = Arrays.asList(refEnt1);
+			Collection<GKInstance> referrers2 = Arrays.asList(refEnt2,refEnt3); 
+			
+			when(molecule1.getReferers("referenceEntity")).thenReturn(referrers1);
+			when(molecule2.getReferers("referenceEntity")).thenReturn(referrers2);
+			Collection<GKInstance> molecules = Arrays.asList(molecule1, molecule2, molecule3, molecule4);
+			when(adaptor.fetchInstanceByAttribute("ReferenceMolecule", "referenceDatabase", "=", "123456")).thenReturn(molecules);
+			
+			updator.updateChebiReferenceMolecules();
+		}
+		catch (Exception e)
+		{
+			fail(e.getMessage());
+			throw new RuntimeException(e);
+		}
+		// If we got this far, it means nothing crashed.
+		assertTrue(true);
+	}
+	
+	@Test
+	public void testChangedIdentifiers() throws InvalidAttributeException, Exception
 	{
-		
 		when(chebiEntity1.getChebiAsciiName()).thenReturn("NAME1");
-		when(chebiEntity2.getChebiAsciiName()).thenReturn("NAME-2");
+		// This mock ChEBI entity will return a different ChEBI ID! It's ID is now the same as molecule2.
+		when(chebiEntity1.getChebiId()).thenReturn("CHEBI:332211");
 		
-		when(chebiEntity1.getChebiId()).thenReturn("CHEBI:112233");
-		when(chebiEntity2.getChebiId()).thenReturn("CHEBI:332211-2");
+		when(chebiEntity2.getChebiAsciiName()).thenReturn("NAME-2");
+		when(chebiEntity2.getChebiId()).thenReturn("CHEBI:332211");
```

Comment:
Just wanted to double check -- should chebiEntity2 and chebiEntity1 above return the same values for a call to getChebiId()?

Reply:
That diff looks... weird. It's comparing line 92 to line 235. The method in which line 235 currently exists was not even in that file prior to this PR, so I think github got confused.

Reply:
Okay, looks like there's no problem then after going over it in person!